okhttp-ws/src/main/java/com/squareup/okhttp/ws/WebSocket.java

```diff
@@ -15,36 +15,28 @@
  */
 package com.squareup.okhttp.ws;
 
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.RequestBody;
 import java.io.IOException;
 import okio.Buffer;
-import okio.BufferedSink;
 
 /** Blocking interface to connect and write to a web socket. */
 public interface WebSocket {
-  /** The format of a message payload. */
-  enum PayloadType {
-    /** UTF8-encoded text data. */
-    TEXT,
-    /** Arbitrary binary data. */
-    BINARY
-  }
+  // Using a "utf-8" charset here is an optimization for use by RequestBody.create().
+  /** A {@link MediaType} indicating that text frames should be used when sending a message. */
+  MediaType TEXT = MediaType.parse("application/vnd.okhttp.websocket+text; charset=utf-8");
+  /** A {@link MediaType} indicating that binary frames should be used when sending a message. */
+  MediaType BINARY = MediaType.parse("application/vnd.okhttp.websocket+binary; charset=utf-8");
```

Comment:
Yeah, the code that adds the charset when it applies one can be surprising. If we do ever need to gracefully tolerate this, the easiest way is to compare `type()` and `subtype()` instead of the entire `MediaType` object. (Though that's cumbersome I admit).